090-integrate-bridge-with-metrics-reporter.md

```diff
@@ -0,0 +1,117 @@
+When this variable is set to true, the Bridge creates the JMX Exporter's JmxCollector class with hard coded configuration.
+This configuration is a JMX metrics exporter YAML distributed as an embedded resource within the Bridge's JAR file.
+
+When deploying the Bridge through the Cluster Operator, a similar configuration is available in the KafkaBridge's CRD:
+
+```sh
+spec:
+  enableMetrics: true
+```
+
+This is different from how the metrics endpoint is enabled in all the other major components deployed by the Cluster Operator.
+The Cluster Operator exposes metrics through the [Prometheus JMX Exporter](https://github.com/prometheus/jmx_exporter), which can be configured using the shared `metricsConfig` schema.
+This schema has a type property that only allows the `jmxPrometheusExporter` value, and a reference to a ConfigMap containing its configuration:
+
+```sh
+metricsConfig:
+  type: jmxPrometheusExporter
+  valueFrom:
+    configMapKeyRef:
+      name: $COMPONENT_NAME-metrics
+      key: metrics-config.yml
+```
+
+At runtime, the above configuration enables a Java agent which exposes Kafka JMX metrics in Prometheus format through an HTTP endpoint on port 9404.
+Note that this agent depends on the [Kafka JMX Reporter](https://github.com/apache/kafka/blob/3.9.0/clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java) plugin, which is enabled by default.
+
+## Motivation
+
+We want to support the Strimzi Metrics Reporter as an alternative way of configuring metrics across all components.
+When deploying configuring major components through the Cluster Operator, we want to provide a consistent user experience, so the Bridge should also support the `metricsConfig` schema. 
+
+## Proposal
+
+The Bridge will support both JMX Prometheus Exporter and Strimzi Metrics Reporter as metrics configuration types.
+It will also provide new configurations for the integration with Cluster Operator.
+
+### Metrics configuration types
+
+The Bridge project will be updated to support multiple metrics configurations types.
+A new `bridge.metrics` property will be available within the application.properties configuration file, and will only accept `jmxPrometheusExporter` and `strimziMetricsReporter`.
+Any other value will raise an error and the application will fail to start with an appropriate error message.
+
+When running in standalone mode with `strimziMetricsReporter`, the user will be able to configure any reporter property using the "kafka." prefix.
+The following example will be provided as a comment in the default application.properties file:
+
+```sh
+bridge.metrics=strimziMetricsReporter
+kafka.metric.reporters=io.strimzi.kafka.metrics.KafkaPrometheusMetricsReporter
+kafka.prometheus.metrics.reporter.listener.enable=false
+kafka.prometheus.metrics.reporter.allowlist=.*
+```
+
+The `KAFKA_BRIDGE_METRICS_ENABLED` environment variable will be deprecated and removed in a future release.
```

Comment:
should we make a clearer plan. I.e. deprecated in 0.32.0 (next release when we have the Strimzi metrics reporter) and removed in 0.33.0 or 0.34.0? Of course the releases can be shifted if we don't get this proposal implemented on time for the 0.32.0 release.

Reply:
Ok, I will propose a schedule so that we can discuss.